jfr-streaming/src/main/java/io/opentelemetry/contrib/jfr/Agent.java

```diff
@@ -0,0 +1,126 @@
+package io.opentelemetry.contrib.jfr;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.metrics.GlobalMeterProvider;
+import io.opentelemetry.api.metrics.MeterProvider;
+import io.opentelemetry.exporter.otlp.metrics.OtlpGrpcMetricExporter;
+import io.opentelemetry.sdk.metrics.SdkMeterProvider;
+import io.opentelemetry.sdk.metrics.SdkMeterProviderBuilder;
+import io.opentelemetry.sdk.metrics.common.InstrumentType;
+import io.opentelemetry.sdk.metrics.view.Aggregation;
+import io.opentelemetry.sdk.metrics.export.PeriodicMetricReader;
+import io.opentelemetry.sdk.metrics.view.InstrumentSelector;
+import io.opentelemetry.sdk.metrics.view.View;
+import io.opentelemetry.sdk.resources.Resource;
+import jdk.jfr.EventSettings;
+import jdk.jfr.consumer.RecordingStream;
+import io.opentelemetry.contrib.jfr.metrics.RecordedEventHandler;
+import io.opentelemetry.contrib.jfr.metrics.HandlerRegistry;
+
+import java.lang.instrument.Instrumentation;
+import java.time.Duration;
+import java.util.concurrent.Executors;
+import java.util.function.Consumer;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import static io.opentelemetry.sdk.metrics.common.InstrumentType.*;
+import static io.opentelemetry.sdk.metrics.data.AggregationTemporality.DELTA;
+
+public class Agent {
+    private static final Logger logger = Logger.getLogger(Agent.class.getName());
+
+    private static final String OTLP_URL = "OTLP_URL";
+    private static final String API_KEY = "API_KEY";
+    private static final String SERVICE_NAME = "SERVICE_NAME";
+    private static final String API_KEY_HEADER = "api-key";
+    private static final String SERVICE_NAME_HEADER = "service.name";
+    private static final String DEFAULT_URL = "http://localhost:4317";
+    private static final String DEFAULT_SERVICE_NAME = "jfr-otlp-bridge";
+    private static final long EXPORT_MILLIS = 2000;
+
+    private static MeterProvider meterProvider;
+
+    public static void agentmain(String agentArgs, Instrumentation inst) {
+      premain(agentArgs, inst);
+    }
+
+    public static void premain(String agentArgs, Instrumentation inst) {
+      meterProvider = GlobalMeterProvider.get();
+      if (meterProvider == null) {
+        configureOpenTelemetry();
+      }
+      startJfrService(agentArgs, inst);
+    }
+
+    public static void startJfrService(String agentArgs, Instrumentation inst) {
```

Comment:
Looks like `Instrumentation` isn't used - can we extract into a non-agent entrypoint, probably `JfrMetrics.enable()` (and ideally `JfrMetrics.disable()` if it's supportable) to allow use programatically? We can just call it from the agent entrypoint if the agent seems useful (and let's publish the shadow jar with `-agent` classifier).

But I'm a bit wary of publishing another javaagent - is it better to add this to the normal opentelemetry-javaagent instead of having yet another agent?

It might keep this PR simpler to remove the agent stuff for now and add it later, either here, or in the otel javaagent.